cmd/launcher/flare.go

```diff
@@ -0,0 +1,331 @@
+	user, err := user.Current()
+	if err != nil {
+		fatal(b, err)
+	}
+	output(b, stdout, "CurrentUser: %s uid: %s\n", user.Username, user.Uid)
+	v := version.Version()
+	jsonVersion, err := json.Marshal(&v)
+	if err != nil {
+		fatal(b, err)
+	}
+	output(b, stdout, "%v\n", string(jsonVersion))
+
+	logger := log.NewLogfmtLogger(b)
+	err = reportGRPCNetwork(
+		logger,
+		serverURL,
+		insecureTLS,
+		insecureGRPC,
+		enrollSecret,
+		certPins,
+		rootPool,
+	)
+	output(b, stdout, "GRPC Connection ...%v\n", err == nil)
+	if err != nil {
+		output(b, fileOnly, "reportGRPCNetwork error: %s", err)
+	}
+
+	err = reportOsqueryProcessInfo(logger, flareSocketPath, b)
+	if err != nil {
+		output(b, fileOnly, "reportOsqueryProcessInfo error: %s", err)
+	}
+	output(b, stdout, "Osqueryi ProcessInfo ...%v\n", err == nil)
+
+	err = reportNotaryPing(logger)
+	if err != nil {
+		output(b, fileOnly, "reportNotaryPing error: %s", err)
+	}
+	output(b, stdout, "Osqueryi Ping Notary ...%v\n", err == nil)
+
+	return nil
+}
+
+type outputDestination int
+
+const (
+	fileOnly outputDestination = iota
+	stdout
+)
+
+func fatal(w io.Writer, err error) {
+	output(w, stdout, "error: %s\n", err)
+	os.Exit(1)
+}
+
+func output(w io.Writer, printTo outputDestination, f string, a ...interface{}) error {
+	if printTo == stdout {
+		fmt.Printf(f, a...)
+	}
+
+	_, err := fmt.Fprintf(w, f, a...)
+	return err
+}
+
+// starts an osqueryd runtime, and then connects an osquery client and runs queries to check and log process info.
+func reportOsqueryProcessInfo(
+	logger log.Logger,
+	socketPath string,
+	output io.Writer,
+) error {
+	logger.Log(
+		"flare", "reportOsqueryProcessInfo",
+		"msg", "creating osquery runner",
+		"socketPath", socketPath,
+	)
+	// create the osquery runtime socket directory
+	if _, err := os.Stat(filepath.Dir(socketPath)); os.IsNotExist(err) {
+		if err := os.Mkdir(filepath.Dir(socketPath), fs.DirMode); err != nil {
+			return errors.Wrap(err, "creating socket path base directory")
+		}
+	}
+
+	// start a osquery runtime
+	runner, err := runtime.LaunchInstance(
+		runtime.WithExtensionSocketPath(socketPath),
+	)
+	if err != nil {
+		return errors.Wrap(err, "creating osquery instance for process info query")
+	}
+	defer func() {
+		if err := runner.Shutdown(); err != nil {
+			logger.Log(
+				"msg", "shutting down runner from reportOsqueryProcessInfo",
+				"err", err,
+			)
+		}
+	}()
+
+	logger.Log(
+		"flare", "reportOsqueryProcessInfo",
+		"msg", "creating osquery-go client",
+		"socketPath", socketPath,
+	)
+
+	// start a client and query it
+	client, err := osquerygo.NewClient(socketPath, 5*time.Second)
+	if err != nil {
+		return errors.Wrapf(err, "creating osquerygo client with socket path %s", socketPath)
+	}
+	defer client.Close()
+
+	logger.Log(
+		"flare", "reportOsqueryProcessInfo",
+		"msg", "running query with osquery-go",
+	)
+
+	const query = `select * from processes where name like '%osqueryd%' OR name like '%launcher%';`
```

Comment:
Doing the data analysis via osquery is a clever way to get the cross platform ability. +100

Reply:
We probably want a simple `ps` output as well, but here I wanted to see if we can stand up and communicate with an osquery process over the extension.